cpp/tests/sampling/mg_uniform_neighbor_sampling.cu

```diff
@@ -143,11 +143,31 @@ class Tests_MGUniform_Neighbor_Sampling
 
     rmm::device_uvector<int32_t> batch_number(random_sources.size(), handle_->get_stream());
 
+    thrust::tabulate(
+      handle_->get_thrust_policy(),
+      batch_number.begin(),
+      batch_number.end(),
+      [rank       = handle_->get_comms().get_rank(),
+       batch_size = uniform_neighbor_sampling_usecase.batch_size] __device__(int32_t index) {
+        return rank + (index / batch_size);
+      });
+
+    size_t num_batches = 1 + handle_->get_comms().get_rank() +
+                         (batch_number.size() / uniform_neighbor_sampling_usecase.batch_size);
+    num_batches = cugraph::host_scalar_allreduce(
+      handle_->get_comms(), num_batches, raft::comms::op_t::MAX, handle_->get_stream());
+
+    rmm::device_uvector<int32_t> unique_batches(num_batches, handle_->get_stream());
+    rmm::device_uvector<int32_t> comm_ranks(num_batches, handle_->get_stream());
+
+    cugraph::detail::sequence_fill(
+      handle_->get_stream(), unique_batches.data(), unique_batches.size(), int32_t{0});
     thrust::tabulate(handle_->get_thrust_policy(),
-                     batch_number.begin(),
-                     batch_number.end(),
-                     [batch_size = uniform_neighbor_sampling_usecase.batch_size] __device__(
-                       int32_t index) { return index / batch_size; });
+                     comm_ranks.begin(),
+                     comm_ranks.end(),
+                     [num_gpus = handle_->get_comms().get_size()] __device__(auto index) {
+                       return index % num_gpus;
+                     });
```

Comment:
Is this logic correct?

After
```
    thrust::tabulate(
      handle_->get_thrust_policy(),
      batch_number.begin(),
      batch_number.end(),
      [rank       = handle_->get_comms().get_rank(),
       batch_size = uniform_neighbor_sampling_usecase.batch_size] __device__(int32_t index) {
        return rank + (index / batch_size);
      });
```
Batch IDs in each GPU will be [rank, (local_number_of_sources + (batch_size - 1)) / batch_size).

Say local_number_of_sources = 100 and batch_size = 10.

GPU 0, batch IDs will be [0, 10)
GPU 1, Batch IDs will be [1, 11).

Isn't `num_batches` the number of unique batch IDs? In the case above, this should be 11 (0, 1, 2, ... , 10), right?

Not sure how
```
    size_t num_batches = 1 + handle_->get_comms().get_rank() +
                         (batch_number.size() / uniform_neighbor_sampling_usecase.batch_size);
    num_batches = cugraph::host_scalar_allreduce(
      handle_->get_comms(), num_batches, raft::comms::op_t::MAX, handle_->get_stream());
```
computes the number of unique batches.

And shouldn't the batch_size be globally enforced for each batch ID? Based on this logic, batch_size will be a local batch_size, and global batch_size for each batch ID will grow with # GPUs. Am I missing something?

Reply:
Yes, this logic is faulty.  I will reconsider.

Reply:
Restructured this code in latest push.  Batch sizes should now be correct except for the last batch which might be small.

Reply:
The new update looks good to me.